C++ generation issue related to std::filesystem on Some platforms

On the Mac platform with a certain configuration CPP Tests are failing for runAllTests.cpp as below. 

The configuration is:

```
Apple clang version 17.0.0 (clang-1700.0.13.5)
Target: arm64-apple-darwin24.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

The error is the following:

```
testRTCpp:
     [echo] Compiling RTCpp tests using gcc
     [exec] runAllTests.cpp:11:21: error: expected namespace name
     [exec]    11 | namespace fs = std::filesystem;
     [exec]       |                ~~~~~^
     [exec] runAllTests.cpp:49:23: error: use of undeclared identifier 'fs'
     [exec]    49 |     auto latestTime = fs::file_time_type::min();
     [exec]       |                       ^
     [exec] runAllTests.cpp:51:30: error: use of undeclared identifier 'fs'
     [exec]    51 |     for (const auto& entry : fs::directory_iterator(directory)) {
     [exec]       |                              ^
     [exec] runAllTests.cpp:54:36: error: use of undeclared identifier 'fs'
     [exec]    54 |             auto currentFileTime = fs::last_write_time(entry);
     [exec]       |                                    ^
     [exec] runAllTests.cpp:131:33: error: use of undeclared identifier 'fs'
     [exec]   131 |         for (const auto& file : fs::directory_iterator(dir)) {
     [exec]       |                                 ^
     [exec] 5 errors generated.

```

There is some discussion of possible solutions here:

https://www.omi.me/blogs/firmware-guides/how-to-fix-error-filesystem-is-not-a-namespace-name?srsltid=AfmBOopZr0ip3zUSlagbCmgI1NH2iYMcsRSmoz3eCwhKBvVnDeHh1Yl3

The implication of the above is that this should be working fine on C++17 and above. It may be a local configuration issue on the Mac (maybe figure out how to install a more recent version, or an earlier version). But we should get to the bottom of this since it will impact other people most likely.

The correct header 
```
#include <filesystem>
```
is already included

The answer should not be to go back to the experimental version

Fixing this will be needed before proper work can continue on other features and PRs. 

If a quick fix cannot be found (that doesn't break other platforms using other gcc versions) then it may be needed to temporarily deactivate this testing (but don't close the issue until the testing is re-activated)

--- a/build/build.testbed.xml
+++ b/build/build.testbed.xml
@@ -282,7 +282,11 @@
   </target>
   
   <target name="testRTCpp">
-    <echo message="Compiling RTCpp tests using gcc"/>
+    <echo message="Compiling RTCpp tests using gcc (if the following says clang it may not work, in which case install a real gcc and symlink it"/>
+    <exec executable="gcc">
+        <arg value="--version"/>    
+    </exec>    
+    
     <exec executable="gcc" dir="testbed_cpp/test" failonerror="true">
         <arg value="-o"/>
         <arg value="runAllTests"/>

--- a/umpleonline/scripts/log.php
+++ b/umpleonline/scripts/log.php
@@ -38,6 +38,8 @@
       else $isSuccess = TRUE;
     }
   }
+  // Force output so far to be emitted
+  flush();
   
   if($isSuccess) {
     echo "<p>Successfully established connection to server\n";
@@ -57,12 +59,27 @@
       $PATH .= ":/usr/local/bin";
     }
     putenv("PATH=$PATH");
+    
+    echo " Path: ";
+    passthru("which java 2>&1");
  
     echo "<p>Dot/Graphviz version: ";
     passthru("dot -V 2>&1");
 
+    echo " Path: ";
+    passthru("which dot 2>&1");
+
+    echo "\n<p>gcc version: ";
+    passthru("gcc --version 2>&1");
+
+    echo " Path: ";
+    passthru("which gcc 2>&1");
+
     echo "\n<p>Docker version: ";
     passthru("docker --version 2>&1");
+
+    echo " Path: ";
+    passthru("which docker 2>&1");
 
     echo "<span style =\"Color: blue\">";
     echo "\n<p> Git commit: ";
@@ -81,7 +98,10 @@
       $isSuccess - FALSE;
     }
   }
-  
+
+  // Force output so far to be emitted
+  flush();
+
   if($isSuccess) {
     echo "<p>Successfully sent log command to server. Results follow\n<pre>\n";  
     socket_set_option($theSocket, SOL_SOCKET, SO_RCVTIMEO,array("sec"=>25,"usec"=>500000) ); 
@@ -112,6 +132,8 @@
     socket_close($theSocket);
   }
 
+  // Force output so far to be emitted, since docker can hang
+  flush();
   echo "<h4 style =\"Color: blue\">Umple collab server status</h4>";
 
   // Obtain the config info and iterate through each config

